src/sdk/widgetHandle.ts

```diff
@@ -185,16 +187,20 @@ export class WidgetHandle {
     state: WidgetState;
     error?: WidgetErrorData;
     resetTrigger?: WidgetResetTrigger;
+    mode?: WidgetMode;
   }) {
     const didStateChange = this.state !== s.state;
     this.response = s.response;
     this.state = s.state;
     if (this.hiddenFormEl && this.e.isConnected !== false) {
       this.hiddenFormEl.value = s.response;
     }
+    if (s.mode) {
```

Comment:
Hmm, maybe the `this.widgetMode` field is not necessary (as it's used right away here)?